### PR TITLE
Ensure that implementation-defined Intl operations access properties …

### LINKED ISSUE #1388: Delegate review: Specify in which order properties are accessed when passing object to implementation-defined operation
From #1383. In implementation-defined operations such as non-ISO calendars (e.g. https://tc39.es/proposal-temporal/#sup-temporal.calendar.prototype.datefromfields) it must be specified in which order properties from these objects are accessed. The objects should be copied and objects containing data properties should be passed to the operations, so that getters are not called more than once.

cc @bmeck

### PR DESCRIPTION
**Pull request by cjtenny**
…in consistent order and only call getters once.

Fixes #1388.

### CODE REVIEW

File: spec/intl.html [hunk 1]
```diff
@@ -1258,6 +1266,8 @@ <h1>Temporal.Calendar.prototype.eraYear ( _dateOrDateTime_ )</h1>
           <emu-alg>
             1. Let _calendar_ be the *this* value.
             1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
+            1. If _dateOrDateTime_ does not have an [[ISOYear]] internal slot, then
```
Comment (ljharb) [inline spec/intl.html:1269]: I’m not sure the spec asks this about something it doesn’t already know is an Object. It seems intuitive that a primitive would neatly fail this test, and i don’t have a specific change in mind, but i wanted to call it out.

Comment (cjtenny) [reply spec/intl.html:1269]: That's a good point. Do you think it'd be better as "`If _dateOrDateTime_ is not an Object or does not have an [[ISOYear]] internal slot, then`"? Or would you prefer the more concise version here?

Comment (ljharb) [reply spec/intl.html:1269]: I think I'll defer to @bakkot, @syg, and @michaelficarra on this one; there's a few other cases in this PR (including "if it has the slot") where it'd apply as well.

Comment (bakkot) [reply spec/intl.html:1269]: I'd probably say `If Type(_dateOrDateTime_) is not Object or _dateOrDateTime_ does not have an [[ISOYear]] internal slot, then`, which is close to the phrasing in [`RequireInternalSlot`](https://tc39.es/ecma262/#sec-requireinternalslot).

Comment (cjtenny) [reply spec/intl.html:1269]: Wilco, thanks

File: polyfill/lib/ecmascript.mjs [hunk 2]
```diff
@@ -1010,7 +1010,7 @@ export const ES = ObjectAssign({}, ES2020, {
     }
     return any ? any : false;
   },
-  PrepareTemporalFields: (bag, fields) => {
+  PrepareTemporalFields: (bag, fields, convert = true) => {
```
Comment (ptomato) [inline polyfill/lib/ecmascript.mjs:1013]: This parameter seems to be no longer used?

**Review by ptomato — APPROVED** [review]
Thanks, all looks correct.

File: polyfill/lib/ecmascript.mjs [hunk 3]
```diff
@@ -1024,7 +1024,7 @@ export const ES = ObjectAssign({}, ES2020, {
         value = defaultValue;
       } else {
         any = true;
-        if (BUILTIN_CASTS.has(property)) {
+        if (convert && BUILTIN_CASTS.has(property)) {
```
Comment (ptomato) [inline polyfill/lib/ecmascript.mjs:1027]: This one has to go too, I'll fix it up

Comment (cjtenny) [reply polyfill/lib/ecmascript.mjs]: oof, where is my mind. thank you.

**Pull request merged**